src/App.tsx

```diff
@@ -1,24 +1,213 @@
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import * as clientService from './api/todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { FilterOptions } from './types/FilterOptions';
 
-const USER_ID = 0;
+const USER_ID = 12166;
+const getFilteredTodos = (todos: Todo[], filterBy: FilterOptions) => {
+  switch (filterBy) {
+    case FilterOptions.All:
+      return todos;
+    case FilterOptions.Active:
+      return todos.filter(todo => !todo.completed);
+    case FilterOptions.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterOptions.All);
+  const filteredTodos = getFilteredTodos(todos, filterBy);
+  const uncompletedTodos = todos.filter(todo => !todo.completed);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorMasage, setErrorMasage] = useState(Errors.allGood);
+  const titleField = useRef<HTMLInputElement>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const reset = () => {
+    setTodoTitle('');
+    setErrorMasage(Errors.allGood);
+    setTemporaryTodo(null);
+    setIsDisabled(false);
+  };
+
+  function loadTodos() {
+    clientService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMasage(Errors.cantGetArray));
+  }
+
+  const addTodo = (
+    title: string, userId: number, completed: boolean,
+  ) => {
+    setIsDisabled(true);
+    setTemporaryTodo({
+      title, userId, completed: false, id: 0,
+    });
+    clientService.addTodo({ title, userId, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => {
+          return [...currentTodos, newTodo];
+        });
+      })
+      .catch((error) => {
+        setErrorMasage(Errors.cantAdd);
+        setTemporaryTodo(null);
+        setIsDisabled(false);
+        throw error;
+      })
+      .then(() => {
+        reset();
+      });
+  };
+
+  const deleteTodo = (id: number) => {
+    return clientService.deleteTodo(id)
+      .then(() => {
+        setTodos(todos.filter((todo) => todo.id !== id));
+      })
+      .catch((error) => {
+        setErrorMasage(Errors.cantDelete);
+        throw error;
+      });
+  };
+
+  const onClearCompleted = () => {
+    todos.filter(todo => todo.completed).map(todo => (
+      deleteTodo(todo.id)
+        .catch((error) => {
+          throw error;
+        })
+        .then(() => {
+          setTodos(uncompletedTodos);
+        })
+    ));
+  };
+
+  // // eslint-disable-next-line react-hooks/exhaustive-deps
+  // useEffect(loadTodos, [USER_ID]);
+  useEffect(loadTodos);
```

Comment:
```suggestion
  useEffect(() => loadTodos(), [USER_ID]);
```
you can make it like this to avoid disabling the linter. and remove commented code)

Reply:
still has
![Снимок экрана 2024-01-13 152657](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/82728301/f9077395-52a0-493f-a81f-107c5164ab48)